src/java.base/share/classes/java/lang/ref/package-info.java

```diff
@@ -154,20 +154,34 @@
  *
  * <li>The dequeuing of a reference to a
  * {@linkplain Cleaner#register(Object object, Runnable action) registered}
- * object, by the Cleaner thread, <i>happens-before</i> the Cleaner thread runs
+ * object, by a Cleaner thread, <i>happens-before</i> that Cleaner thread runs
```

Comment:
Are we committing to this? I.e. that there will always be a single Cleaner thread? (i.e. one could imagine a situation where you have one polling thread which executes the cleaning action on some other thread?)

Reply:
I'll revert this back to "_the_ Cleaner thread".